src/util/Util.js

```diff
@@ -481,12 +482,11 @@ class Util extends null {
    * @returns {Collection}
    */
   static discordSort(collection) {
-    return collection.sorted(
-      (a, b) =>
-        a.rawPosition - b.rawPosition ||
-        parseInt(b.id.slice(0, -10)) - parseInt(a.id.slice(0, -10)) ||
-        parseInt(b.id.slice(10)) - parseInt(a.id.slice(10)),
-    );
+    return collection.sorted((a, b) => {
+      const aId = BigInt(a.id);
+      const bId = BigInt(b.id);
+      return a.rawPosition - b.rawPosition || Number(a instanceof GuildChannel ? aId - bId : bId - aId);
```

Comment:
We should avoid converting to BigInt unless both `rawPosition`s are the same, otherwise it'll cause some unneeded overhead.